src/base-client-helpers_test.ts

```diff
@@ -0,0 +1,141 @@
+import { assertEquals } from "https://deno.land/std@0.185.0/testing/asserts.ts";
+import { _internals, getUserAgent } from "./base-client-helpers.ts";
+import { serializeData } from "./base-client-helpers.ts";
+import { assertSpyCalls, stub } from "./dev_deps.ts";
+
+Deno.test(_internals.getModuleVersion.name, async (t) => {
+  await t.step(
+    "should return the version if the module is sourced from deno.land",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("https://deno.land/x/deno_slack_api@2.1.0/mod.ts)");
+      });
+
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, "2.1.0");
+      } finally {
+        getModuleUrlStub.restore();
+      }
+    },
+  );
+
+  await t.step(
+    "should return the unknown if the module is not sourced from deno.land",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("file:///hello/world.ts)");
+      });
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, undefined);
+      } finally {
+        getModuleUrlStub.restore();
+      }
+    },
+  );
+
+  await t.step(
+    "should return the unknown if the module version is invalid",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("https://deno.land/x/deno_slack_sdk@2.1.0/mod.ts)");
+      });
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, undefined);
+      } finally {
+        getModuleUrlStub.restore();
+      }
+    },
+  );
+});
+
+Deno.test(getUserAgent.name, async (t) => {
+  await t.step(
+    "should return the user agent with expected output",
```

Comment:
I'm a nitpicker when it comes to test names, but they are important. A test name can convey a lot of information about a program's behaviour without ever having to look at code. From the perspective of a new contributor unfamiliar with the project, good test names make it easier to ramp up, making it easier to contribute.

Let's be clear what 'expected output' means - you as the developer of this feature know what that means, but me or someone else does not. I would suggest instead:

```suggestion
    "should return the user agent with deno version, OS name and undefined deno-slack-api version",
```

Reply:
Great point, I did overlook the test names here 💯